packages/common/src/s3.ts

```diff
@@ -1,7 +1,33 @@
-import { S3Client } from '@aws-sdk/client-s3';
+import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { createWriteStream } from 'fs';
+import path from 'path';
+import { Readable } from 'stream';
 
-export const getClient = (region: string) => {
+export const getS3Client = (region: string) => {
 	return new S3Client({
 		region,
 	});
 };
+
+export const getFile = async (
+	client: S3Client,
+	bucket: string,
+	key: string,
+	workingDirectory: string,
+) => {
+	try {
+		const destinationPath = `${workingDirectory}/${path.basename(key)}`;
+		const data = await client.send(
+			new GetObjectCommand({
+				Bucket: bucket,
+				Key: key,
+			}),
+		);
+		(data.Body as Readable).pipe(createWriteStream(destinationPath));
```

Comment:
I think you might need to wrap this in a promise otherwise it will return instantly before the file has finished writing. You should be able to check this by downloading a big file and making sure that it hangs here rather than immediately logging 'successfully retrieved file from S3 into '

Reply:
Good spot 👍